internal/controller/authenticator/webhookcachefiller/webhookcachefiller.go

```diff
@@ -63,18 +108,44 @@ func (c *controller) Sync(ctx controllerlib.Context) error {
 		return fmt.Errorf("failed to get WebhookAuthenticator %s/%s: %w", ctx.Key.Namespace, ctx.Key.Name, err)
 	}
 
-	webhookAuthenticator, err := newWebhookAuthenticator(&obj.Spec, os.CreateTemp, clientcmd.WriteToFile)
-	if err != nil {
-		return fmt.Errorf("failed to build webhook config: %w", err)
+	conditions := make([]*metav1.Condition, 0)
+	specCopy := obj.Spec.DeepCopy()
+	var errs []error
+
+	certPool, conditions, tlsBundleOk := c.validateTLSBundle(specCopy.TLS, conditions)
```

Comment:
I wonder if this block of code could live in a func, this would help separate it out.

Reply:
These validation functions do not seem to care about the `conditions` array that is passed into them. I get that `conditions` is being used to accumulate all of the conditions, it just seems odd to both pass it in and return it when the validation function only ever returns a single condition.

Reply:
I'm open to this thought.  Presently we have this pattern in several controllers, not just this one, so its not a deviation from existing code.